python/kvikio/_lib/libnvcomp.pyx

```diff
@@ -57,305 +63,184 @@ class pyNvcompType_t(Enum):
+        return self._impl.get_compressed_output_size(
+            <uint8_t*><uintptr_t>comp_buffer.data.ptr
+        )
 
-    def configure(self, in_bytes, temp_bytes, out_bytes):
-        cdef uintptr_t temp_bytes_ptr = Array(temp_bytes).ptr
-        cdef uintptr_t out_bytes_ptr = Array(out_bytes).ptr
-        self.c.configure(
-            <size_t>in_bytes,
-            <size_t*>temp_bytes_ptr,
-            <size_t*>out_bytes_ptr)
 
-    def compress_async(
+cdef class _ANSManager(_nvcompManager):
+    def __cinit__(
         self,
-        in_arr,
-        in_bytes,
-        temp_arr,
-        temp_bytes,
-        out_arr,
-        out_bytes,
-        uintptr_t stream=0
+        size_t uncomp_chunk_size,
+        user_stream,
+        const int device_id,
     ):
-        cdef uintptr_t in_ptr = Array(in_arr).ptr
-        cdef uintptr_t temp_ptr = Array(temp_arr).ptr
-        cdef uintptr_t out_ptr = Array(out_arr).ptr
-        cdef uintptr_t out_bytes_ptr = Array(out_bytes).ptr
-        self.c.compress_async(
-            <void*>in_ptr,
-            <size_t>in_bytes,
-            <void*>temp_ptr,
-            <size_t>temp_bytes,
-            <void*>out_ptr,
-            <size_t*>out_bytes_ptr,
-            <cudaStream_t>stream)
-
-cdef class _LZ4Decompressor:
-    cdef __LZ4Decompressor* d
-
-    def __cinit__(self):
-        self.d = new __LZ4Decompressor()
-
-    def __dealloc__(self):
-        del self.d
+        self._impl = <nvcompManagerBase*>new ANSManager(
+            uncomp_chunk_size,
+            <cudaStream_t><void*>0,  # TODO
+            device_id
+        )
 
-    cpdef configure(self, in_arr, in_bytes, temp_bytes, out_bytes, uintptr_t stream=0):
-        cdef uintptr_t in_ptr = Array(in_arr).ptr
-        cdef uintptr_t temp_bytes_ptr = Array(temp_bytes).ptr
-        cdef uintptr_t out_bytes_ptr = Array(out_bytes).ptr
-        self.d.configure(
-            <void*>in_ptr,
-            <size_t>in_bytes,
-            <size_t*>temp_bytes_ptr,
-            <size_t*>out_bytes_ptr,
-            <cudaStream_t>stream)
 
-    def decompress_async(
+cdef class _BitcompManager(_nvcompManager):
+    def __cinit__(
         self,
-        in_arr,
-        in_bytes,
-        temp_arr,
-        temp_bytes,
-        out_arr,
-        out_bytes,
-        uintptr_t stream=0
+        nvcompType_t data_type,
+        int bitcomp_algo,
+        user_stream,
+        const int device_id
     ):
-        cdef uintptr_t in_ptr = Array(in_arr).ptr
-        cdef uintptr_t temp_ptr = Array(temp_arr).ptr
-        cdef uintptr_t out_ptr = Array(out_arr).ptr
-        self.d.decompress_async(
-            <void*>in_ptr,
-            <size_t>in_bytes,
-            <void*>temp_ptr,
-            <size_t>temp_bytes,
-            <void*>out_ptr,
-            <size_t>out_bytes,
-            <cudaStream_t>stream)
+        self._impl = <nvcompManagerBase*>new BitcompManager(
+            <nvcompType_t>data_type,
+            <int>bitcomp_algo,
```

Comment:
As these already have these types, can skip the cast. This can be applied in a few other places

```suggestion
            data_type,
            bitcomp_algo,
```